src/index.rs

```diff
@@ -228,17 +230,29 @@ impl Index {
     let genesis_block_coinbase_transaction =
       options.chain().genesis_block().coinbase().unwrap().clone();
 
+    let p2p_url = match Url::parse(&rpc_url) {
+      Ok(url) => url.host().unwrap().to_string(),
+      Err(_) => rpc_url
+        .split(':')
+        .collect::<Vec<_>>()
+        .first()
+        .unwrap()
+        .to_string(),
+    };
```

Comment:
Not sure what the best way to parse the url is. By default it is `127.0.0.1:8332/wallet/ord`, which doesn't parse as a url, and in tests it is `http://127.0.0.1:...` which won't parse with a `split(':')`. Maybe regex?